internal/controller/nimservice_controller.go

```diff
@@ -133,6 +134,18 @@ func (r *NIMServiceReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 	logger.Info("Reconciling", "NIMService", nimService.Name)
 
+	if nimService.Spec.MultiNode != nil && nimService.Annotations != nil {
```

Comment:
annotations could be nil initially, we would need to initialize the map for the first time.

Reply:
also, this should be done in the platform specific reconciler where we render the LWS resource. That way we can consider the case of auto GPU assignment too based on the tp size in the optimized profile, otherwise there would be mismatch and we end up setting this annotation to `0`.